### PR TITLE
@RunWith(SWTBotJunit4ClassRunner) to capture screenshot on testStage() error

### PR DESCRIPTION
**Pull request by briandealwis**

### CODE REVIEW

**Comment by chanseokoh** [comment]
Does this (just adding this annotation) magically make the test capture screenshots? Moreover, this is not an integration test, so what I actually see locally is just the "Welcome" screen.

**Comment by briandealwis** [comment]
It captures screenshots on failure, which will then be uploaded to our travis_builds bucket.  I forgot about that dratted welcome screen though.

File: plugins/com.google.cloud.tools.eclipse.swtbot/src/com/google/cloud/tools/eclipse/swtbot/SwtBotWorkbenchCaretaker.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+/** Take screenshot after test failure, and clean up the workbench. */
+public class SwtBotWorkbenchCaretaker extends TestWatcher {
+  private static final Logger logger = Logger.getLogger(SwtBotWorkbenchCaretaker.class.getName());
+
+  @Override
+  protected void starting(Description description) {
+    try {
+      SwtBotWorkbenchActions.closeWelcome(new SWTWorkbenchBot());
+    } catch (WidgetNotFoundException ex) {
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.swtbot/src/com/google/cloud/tools/eclipse/swtbot/SwtBotWorkbenchCaretaker.java]: In cases like this, I think we usually name the exception variable "ignored" which will turn off some static analysis checks.

File: plugins/com.google.cloud.tools.eclipse.swtbot/src/com/google/cloud/tools/eclipse/swtbot/SwtBotWorkbenchCaretaker.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import org.junit.runner.Description;
+
+/** Take screenshot after test failure, and clean up the workbench. */
+public class SwtBotWorkbenchCaretaker extends TestWatcher {
+  private static final Logger logger = Logger.getLogger(SwtBotWorkbenchCaretaker.class.getName());
+
+  @Override
+  protected void starting(Description description) {
+    try {
+      SwtBotWorkbenchActions.closeWelcome(new SWTWorkbenchBot());
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.swtbot/src/com/google/cloud/tools/eclipse/swtbot/SwtBotWorkbenchCaretaker.java]: Doesn't seem to work on Travis.

BTW, is there a way to wait until the Welcome view is closed? Seems like it is closed some time after the actual test method runs.

Comment (briandealwis) [reply plugins/com.google.cloud.tools.eclipse.swtbot/src/com/google/cloud/tools/eclipse/swtbot/SwtBotWorkbenchCaretaker.java]: I blithely assumed from the log message ([_ Check the error message in the Console View_](https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/3107#issuecomment-394708059)) that we had a workbench open here, and so wanted to take a screenshot of the _Package Explorer_ or _Project Explorer_ to see what it showed in the console view. But the workbench isn't open!

Comment (chanseokoh) [reply plugins/com.google.cloud.tools.eclipse.swtbot/src/com/google/cloud/tools/eclipse/swtbot/SwtBotWorkbenchCaretaker.java]: Yeah, to be frank, I was thinking taking a screenshot won't help here.

**Pull request closed without merging**